src/libraries/System.Reflection.Emit/tests/AssemblyBuilderTests.cs

```diff
@@ -424,7 +423,17 @@ void Invoke_Private_SameAssembly_ThrowsMethodAccessException()
 	    Assert.Throws<MethodAccessException>(() => d ());
 	}
 
-
+    [Fact]
+    public void DefineDynamicAssembly_InternalAssemblyLocationIsEmpty()
+    {
+        AssemblyBuilder assembly = Helpers.DynamicAssembly();
+        object internalAssemblyBuilder = assembly.GetType()
+                .GetProperty("InternalAssembly", BindingFlags.NonPublic | BindingFlags.Instance).GetValue(assembly);
+        object locationObj = internalAssemblyBuilder.GetType().GetProperty("Location", BindingFlags.Public | BindingFlags.Instance)
+                .GetValue(internalAssemblyBuilder);
+        string location = Assert.IsType<string>(locationObj);
```

Comment:
Would it be possible to rewrite the test to not use private reflection?
The test is valid regardless of how the builder is implemented, but now it would fail if we changed the implementation (to rename/remove the internal property).
I think it should be possible to call something like `AssemblyLoadContext.Default.Assemblies` and find the one just defined and validate it there.

Reply:
thanks for the clue